---
title: "LD calculation for 282 panel"
output: html_notebook
date: 08-31-2018
author: "Jinliang Yang"
---

## LD 

```{bash, eval=FALSE}
cd /common/jyanglab/jyang21/dbcenter/HapMap3/set282
plink -bfile allchr_bisnp_n282_snpid_maf001_geno1 --r2 --ld-window 100 --ld-window-kb 100 --ld-window-r2 0 --out allchr_bisnp_n282_snpid_maf001_geno1
```

```{r setup, include=TRUE}
#setwd("~/Documents/Github/pvpDiallel/")
knitr::opts_knit$set(root.dir=normalizePath('../../'))
```


## Calculate LD

```{r, eval=FALSE}
library("data.table")

df <- fread("/common/jyanglab/jyang21/dbcenter/HapMap3/set282/allchr_bisnp_n282_snpid_maf001_geno1.ld", data.table=FALSE)

BINSIZE = 100
df$dist <- df$BP_B -df$BP_A
df$bin <- round(df$dist/BINSIZE, 0)

library(plyr)

df2 <- ddply(df, .(bin), summarise,
      meanr2 = mean(R2))

write.table(df2, "cache/ld_in_100bp_bin.csv", sep=",", row.names=FALSE, quote=FALSE)

```


## Plot LD
```{r}
ld <- read.csv("cache/ld_in_100bp_bin.csv")

plot(ld$bin, ld$meanr2)

abline(h=0.02, col="red")
```














```{r}
library("data.table")

maf <- fread("largedata/allchr_bisnp_n282_snpid_maf001_geno5.frq", header=TRUE)
lmiss <- fread("largedata/allchr_bisnp_n282_snpid_maf001_geno5.lmiss", header=TRUE)


pdf("graphs/maf_lmiss.pdf", width = 10, height=5)
par(mfrow=c(1,2))
hist(maf$MAF, breaks=50, col="#cdb79e", main="MAF (SNP = 28,014,740)", xlab="Minor Allele Freq")
#abline(v=0.01, lty=2, col="black", lwd=3)
abline(v=0.05, lty=2, col="red", lwd=3)

hist(lmiss$F_MISS, breaks=35, col="#cdb79e", main="Missingness (SNP = 28,014,740)", xlab="Missing Rate")
#abline(v=0.6, lty=2, col="red", lwd=3)
#abline(v=0.05, lty=2, col="red", lwd=3)
dev.off()

```

Raghu's LD decay analysis:

```{bash}
cd /work/jyanglab/raghu/jyang/zmhmp/cache
 plink --bfile allchr_0.001_0.2 --ld-window-kb 100 --ld-window 100  --r2 --ld-window-r2 0  --out all_chr_LD
```


```{r,eval=FALSE}
cd /work/jyanglab/raghu/jyang/zmhmp/cache
library("data.table")
df <- fread ("all_chr_LD.ld",header=T)
#from n=1221021
df<-df[,-c(3,6)]
head(df)
df$dist <- df$BP_B - df$BP_A 
#BINSIZE=100
#BINSIZE=50
BINSIZE=25
df$bin=round(df$dist/BINSIZE,0)
head(df)
df<-df[,-c(1:4)]
head(df)
library(plyr)
df2 <- ddply(df,.(bin),summarise, meanr2=mean(R2))
head(df2)
tail(df2)
write.table(df2,"ld_binsize_50.csv")

#plot with 2000 bins
pdf("ld_binsize_50.pdf")
library("ggplot2")
ggplot(df2, aes(df2$bin, df2$meanr2)) + geom_point()+ geom_smooth(se=F) 
dev.off()


#Zoom-in
pdf("ld_binsize_50_zoomin.pdf")
library("ggplot2")
df3 <- subset(df2,bin<100)
ggplot(df3, aes(df3$bin, df3$meanr2)) + geom_point()+ geom_smooth(se=F) 
dev.off()
#still could not see many points above r2=0.1

#Ignore mean r2 and plot just r2 against distance for the first 10 kb.
pdf("ld_in_10kb.pdf")
library("ggplot2")
df4 <- subset(df,dist<10)
ggplot(df, aes(df$dist, df$R2)) + geom_smooth(se=F) 
dev.off()



pdf("ld_binsize1bp_10kb_window.pdf")
ggplot(df5, aes(df5$bin, df5$meanr2)) + geom_point()+ geom_smooth(se=F) 
dev.off()

```


```{bash, eval=FALSE}
cd /work/jyanglab/raghu/jyang/zmhmp/cache/pruned_0.001_0.2/
plink --bfile all_chr_pruned001_0.2 --recode vcf --out all_chr_pruned001_0.2

vcftools --vcf all_chr_pruned001_0.2.vcf --SNPdensity 100,000 --out all_chr_pruned001_0.2_density100kb
```

#SNP density visualization (not very helpful chunk)
```{r,eval=F}
setwd("~/Ziegler_mineral_GWAS/data")
snpden=read.csv("SNP_density.csv",header=T)
library(ggplot2)
ggplot(snpden, aes(snpden$CHROM,snpden$SNP_COUNT)) + geom_point()+
scale_x_continuous(breaks = seq(min(snpden$CHROM), max(snpden$CHROM)))

ggplot(snpden, aes(snpden$CHROM,snpden$variantsperKb)) + geom_point()+
  scale_x_continuous(breaks = seq(min(snpden$CHROM), max(snpden$CHROM)))

ggplot(snpden, aes(snpden$CHROM,snpden$variantsperKb)) + geom_point()+
  scale_x_continuous(breaks = seq(min(snpden$CHROM), max(snpden$CHROM)))

ggplot(snpden, aes(snpden$BIN_START,snpden$SNP_COUNT)) + geom_point()+
scale_x_continuous(breaks = seq(min(snpden$BIN_START), max(snpden$BIN_START)))

ggplot(snpden, aes(snpden$BIN_START,snpden$variantsperKb)) + geom_point() +
scale_x_continuous(breaks = seq(min(snpden$BIN_START), max(snpden$BIN_START)))

ggplot(snpden, aes(snpden$CHROM,snpden$BIN_START)) + geom_point() +
scale_x_continuous(breaks = seq(min(snpden$CHROM), max(snpden$CHROM)))


```


#SNP density visualization (with manhattan plots)
```{r}

setwd("~/Ziegler_mineral_GWAS/data")

pdf("~/Ziegler_mineral_GWAS/graphs/manhattan_snpdensity.pdf")
snpden=read.csv("SNP_density.csv",header=T)
library(ggplot2)

# Load the library
library(qqman)

# Make the Manhattan plot on the gwasResults dataset
manhattan(snpden, chr="CHROM", bp="BIN_START", p="variantsperKb",col = c("gray10", "gray60"), ylab="variantsperKb")
abline(lm(snpden$variantsperKb~snpden$BIN_START),col="red")

#pdf("~/Ziegler_mineral_GWAS/graphs/manhattan_snpdensity_chr1.pdf")
#manhattan(subset(snpden, CHROM == 1),chr="CHROM", bp="BIN_START", p="variantsperKb",ylab="variantsperKb")
#abline(lm(snpden$variantsperKb~snpden$BIN_START),col="red")
#manhattan(snpden, chr="CHROM", bp="BIN_START", p="SNP_COUNT",col = c("gray10", "gray60"), ylab="variantsperKb")

dev.off()

```

#LD-decay (with 1bp bins) over 10_kb genomic interval

```{r,eval=FALSE}

#on remote
library("ggplot2")
pdf("1bp_bins_10kb-window.pdf")
df4 <- subset(df,dist<10000)
BINSIZE=1
df4$bin=round(df4$dist/BINSIZE,0)
library(plyr)
df5 <- ddply(df4,.(bin),summarise, meanr2=mean(R2))
write.csv(df5,"1bp_bins_10kb-window.csv")

#locally
setwd("/Users/rkastooriramamurth2/Ziegler_mineral_GWAS/data")
df5 <- read.csv("1bp_bins_10kb-window.csv",header=T)
p1 <- ggplot(df5, aes(df5$bin, df5$meanr2)) + geom_point()+ geom_smooth(se=F) 
p1 + geom_hline(yintercept=0.055, linetype="dashed", color = "red")
dev.off()
```
































